raiden/messages/synchronization.py

```diff
@@ -49,10 +48,8 @@ def from_event(cls, event: SendMessageEvent) -> "Processed":
         return cls(message_identifier=event.message_identifier, signature=EMPTY_SIGNATURE)
 
     def _data_to_sign(self) -> bytes:
-        return pack_data(
-            (self.cmdid.value, "uint8"),
-            (b"\x00" * 3, "bytes"),  # padding
-            (self.message_identifier, "uint64"),
+        return bytes([self.cmdid.value, 0, 0, 0]) + self.message_identifier.to_bytes(
```

Comment:
Using python built-ins is always going to be faster than `pack_data` due to the way `map_abi_data` is implemented in [web3.py](https://github.com/ethereum/web3.py/blob/a40aa941280a8b8de0264340bed820651fad95b3/web3/_utils/abi.py#L755).

It's a good change.

Why not also abstract it into a function and replace similar places like this?

For example [here](https://github.com/raiden-network/raiden/blob/b435506037b52407533355112555a2bc98b38552/raiden/messages/healthcheck.py#L27) and [here](https://github.com/raiden-network/raiden/blob/b435506037b52407533355112555a2bc98b38552/raiden/messages/healthcheck.py#L44) and other places .. (you can grep them if you need to)

Reply:
I don't think there is anything wrong with `pack_data` except for it's speed. So I only changed the cases that were visible in the flame graph.

I'm also not sure how the extracted functions should look. `bytes([self.cmdid.value, 0, 0, 0])` is the only commonly shared part and is not really worth it's own function.

Reply:
Yeah there is nothing wrong except for speed. I only suggest it since you found out that it's a bottleneck through the flamegraph.

Due to the less generic nature of not using `map_abi_data` extracted function may be more difficult to do indeed.

I leave it up to you, since you are checking the graphs to see where the bottlenecks are. Approving the PR anyway so that you are not blocked.

Note that an integration test failed. Not sure if it's related: https://circleci.com/gh/raiden-network/raiden/100864#tests/containers/9